### ISSUE TITLE
Enhancement: calcite-slider - custom tick label formatters

### ISSUE THREAD

**Issue by hccampos** [Esri/calcite-design-system#2584]
### Description

For certain use cases the default tick labels don't work well. For example, one may want to display textual labels rather than numerical ones.

It would be great to be able to format slider tick labels. This is already possible in the JS API `Slider` widget by using a `labelFormatFunction`: https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Slider.html#labelFormatFunction

#### Which Component

- `<calcite-slider>`

#### Example Use Case

- Settings slider in the Web SceneViewer
![Screen Shot 2021-07-21 at 10 32 38](https://user-images.githubusercontent.com/3759114/126457859-854175f4-cc63-446c-9643-9c5bafb421f1.jpg)

- Time sliders as in the `Daylight` slider of the JS API:
![Screen Shot 2021-07-21 at 10 33 30](https://user-images.githubusercontent.com/3759114/126457982-134a9b53-d38b-40d3-b255-2293ec431185.jpg)

<sub>SVCH</sub>

**Comment by richiecarmichael** [comment]
> For certain use cases the default tick labels don't work well.

I would argue that the default tick labels don't work well for **_most_** use cases. For example:
- 1669231079644 to "11/23/2022"
- 1669231079644 to "11:18am"
- 0, 1, 2 to "low", "medium and "high"
- 1000000 to 1M
- 1000000 to $1,000,000

This would be really useful for [#46032](https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/46032).

**Comment by ubatsukh** [comment]
@driskull as we discussed yesterday this is a high priority issue for us! MV needs Range Slider for multidimensional image services and the API implementation cannot proceed until the calcite slider has support for this.

**Comment by jcfranco** [comment]
Closing in favor of https://github.com/Esri/calcite-design-system/issues/4004.
